services/pull/review.go

```diff
@@ -197,6 +198,70 @@ func CreateCodeComment(ctx context.Context, doer *user_model.User, gitRepo *git.
 	return comment, nil
 }
 
+// generatePatchForUnchangedLine creates a patch showing code context for an unchanged line
+func generatePatchForUnchangedLine(gitRepo *git.Repository, commitID, treePath string, line int64, contextLines int) (string, error) {
```

Comment:
It seems the two functions are almost duplicated.

Reply:
address in other comment here: https://github.com/go-gitea/gitea/pull/35662#discussion_r2439162694